From a01836c2dba134661534e745d4b0301f5c507b35 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Wed, 21 Aug 2024 10:39:17 +0300
Subject: [PATCH] [FSW-8566] fs1: bloader: copy preboot from misc arc ccm

At the moment, when working with StandAlone Preboot, a fresh copy of
preboot is stored in the MNG.ICCM and copied to SPSRAM after every
reset by the boot loader.

Since MNG.ICCM is used for MNG application, and this application
is about to be tested on the palladium, we need an alternative
location for the preboot image.

Since MISC.ICCM is not used, we can store the image there.

In addition, additional location for large preboot app is changed
to MISC.DCCM

Change-Id: I2965631fe1c86d2b73fb8833fa093d1a73554e12
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/fs1/arc/palad.c b/arch/fs1/arc/palad.c
index e4da780..df14586 100644
--- a/arch/fs1/arc/palad.c
+++ b/arch/fs1/arc/palad.c
@@ -230,11 +230,11 @@
 	u32 val;
 
 	/*
-	 * Clean copy of the application is stored in MNG ICCM. In case the application binary
-	 * exceeds MNG ICCM size (256kb), the rest is stored in MISC ICCM.
-	 * An efuse bit is controlling whether the application binary is larger than MNG ICCM size.
+	 * Clean copy of the application is stored in MISC ICCM. In case the application binary
+	 * exceeds MISC ICCM size (256kb), the rest is stored in MISC DCCM.
+	 * An efuse bit is controlling whether the application binary is larger than MISC ICCM size.
 	 */
-	fs1_mem_copy(ARC_MNG_ICCM_FULL_BASE, PREBOOT_TEXT_BASE_FULL, ARC_MNG_ICCM_SIZE, false);
+	fs1_mem_copy(ARC_MISC_ICCM_FULL_BASE, PREBOOT_TEXT_BASE_FULL, ARC_MISC_ICCM_SIZE, false);
 
 	err = preboot_fuse_read(IO_EFUSE_PLDM_CFG_0_OFFSET, &val);
 	if (err)
@@ -245,8 +245,8 @@
 	if (!val)
 		return;
 
-	fs1_mem_copy(ARC_MISC_ICCM_FULL_BASE, PREBOOT_TEXT_BASE_FULL + ARC_MNG_ICCM_SIZE,
-		     ARC_MISC_ICCM_SIZE, false);
+	fs1_mem_copy(ARC_MISC_DCCM_FULL_BASE, PREBOOT_TEXT_BASE_FULL + ARC_MISC_ICCM_SIZE,
+		     ARC_MISC_DCCM_SIZE, false);
 }
 
 static void btldr_copy_bootinfo(void)
